select.js

```diff
@@ -1,14 +1,39 @@
-define(function() {
-    var init = function() {
-        $('body').on('change', '.c-select', function() {
-            var $container = $(this);
-            var value = $container.find('option:selected').text();
-
-            $container.find('.c-select__value').text(value);
-        });
-    };
+define(['$'], function($) {
+    //
+    // Component constructor
+    //
+    var Select = function Select($el, options) {
+        // Component instance properties. The `$` hash (by convention) caches
+        // the root node and any sub-components.
+        this.value = '';
+        this.dom = {
+            $root: $el,
+            $value: $el.find('.c-select__value')
+        };
+
+        // Event handlers should use $.proxy to preserve access to the instance
+        // as `this` within the handler.
+        this.dom.$root.on('change', $.proxy(this.update, this));
+
+        // Immediately invoke any methods needed to set up initial state.
+        this.update();
+    }
+
+    //
+    // Update the component’s visible value when the underlying select element’s
+    // value changes.
+    //
+    Select.prototype.update = function update() {
+        this.value = this.dom.$root.find('option:selected').text();
+        this.dom.$value.text(this.value);
+        this.dom.$root.trigger('select:update'); // notify subscribers
```

Comment:
That sounds like a good idea. Another way to help figure this out is to look at some use cases for inter-component communication. The classic one would be something like attaching a tooltip to something arbitrary, like a button on a form field:

```html
{@c-text-input type="text" label="CVC"}
    {:secondaryAction}
        {! This should open a tooltip. How? Is there some declarative way? !}
        {@c-icon-button icon="help" /}
{/c-text-input}
```

One example from another framework: Ember allows components to send messages to the parent view controller via an `action` param: http://guides.emberjs.com/v1.10.0/components/sending-actions-from-components-to-your-application/.

Polymer seems to have used a number of methods in its beta period including [observers/data binding and events](https://www.polymer-project.org/0.5/articles/communication.html#events), but now seems to [not talk about events much in version 1.0](https://www.polymer-project.org/1.0/docs/start/quick-tour.html).